src/Models/Services/Utils.ts

```diff
@@ -381,6 +393,187 @@ export function rgbToHex(r, g, b) {
     return '#' + componentToHex(r) + componentToHex(g) + componentToHex(b);
 }
 
+export async function parseModel(modelJson: string) {
+    const modelParser = createParser(
+        ModelParsingOption.PermitAnyTopLevelElement
+    );
+    try {
+        await modelParser.parse([modelJson]);
+    } catch (err) {
+        if (err.name === 'ParsingException') {
+            return err._parsingErrors
+                .map((e) => `${e.action} ${e.cause}`)
+                .join('\n');
+        }
+
+        return err.message;
+    }
+}
+
+// Store OAT-data
+export const storeEditorData = (oatEditorData: ProjectData) => {
+    localStorage.setItem(OATDataStorageKey, JSON.stringify(oatEditorData));
+};
+
+// Get stored OAT-data
+export const getStoredEditorData = () => {
```

Comment:
can we have this return a typed object?

Reply:
+1, lets make sure the data that is in local storage has a type defined at least, and make sure we reference it